pkg/image/importer/importer.go

```diff
@@ -299,9 +319,35 @@ func formatRepositoryError(repository *importRepository, refName string, refID s
 	return
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image, cache *imageStreamImporterCache) error {
+	bs := repo.Blobs(ctx)
+
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+
+		if layerSize, ok := cache.LayerSizeCache[layer.Name]; ok {
+			size += layerSize
+			continue
+		}
+
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
I have a check on the lack of size:
https://github.com/openshift/origin/pull/9115/files#diff-c85ad4ce0adaea09a2751949b944a92bR479
https://github.com/openshift/origin/pull/9115/files#diff-c85ad4ce0adaea09a2751949b944a92bR529

but I'll add an additional check of the size of each layer.